### DISCUSSION TITLE
Tree view with icons

### DISCUSSION THREAD

**Discussion by CSharpDummy** [zauberzeug/nicegui#2501] [Q&A] [answered]
### Question

Is it possible to add icons to the treeview component? If so, how?

**Comment by falkoschindler** [comment]
Hi @CSharpDummy,

You can simply add icons like this:
```py
ui.tree([
    {'id': 'numbers', 'icon': 'pin', 'children': [{'id': '1'}, {'id': '2'}]},
    {'id': 'letters', 'icon': 'abc', 'children': [{'id': 'A'}, {'id': 'B'}]},
], label_key='id')
```

**Reply by CSharpDummy** [reply]
That works! Is there a way to change the color?

**Reply by falkoschindler** [reply, marked as answer]
I don't immediately see an API for it. But you can use some custom CSS:

```py
ui.add_head_html('''
    <style>
        .my-tree .q-tree__icon {
            color: orange;
        }
    </style>
''')
ui.tree([
    {'id': 'numbers', 'icon': 'pin', 'children': [{'id': '1'}, {'id': '2'}]},
    {'id': 'letters', 'icon': 'abc', 'children': [{'id': 'A'}, {'id': 'B'}]},
], label_key='id').classes('my-tree')
```
<img width="123" alt="Screenshot 2024-02-14 at 20 38 17" src="https://github.com/zauberzeug/nicegui/assets/5767091/87ae3a6c-6fac-4b58-b01f-799fed4f074a">

**Reply by CSharpDummy** [reply]
I found that there is property iconColor to set the color. In any case it works, thx!

**Reply by falkoschindler** [reply]
@CSharpDummy I can't find such property. Can you share code or a link?

**Reply by CSharpDummy** [reply]
@falkoschindler it's here:

https://quasar.dev/vue-components/tree#nodes-model-structure

**Reply by falkoschindler** [reply]
Oh, great! So the code would be like this:
```py
ui.tree([
    {'id': 'numbers', 'icon': 'pin', 'iconColor': 'orange', 'children': [{'id': '1'}, {'id': '2'}]},
    {'id': 'letters', 'icon': 'abc', 'iconColor': 'orange', 'children': [{'id': 'A'}, {'id': 'B'}]},
], label_key='id')
```
Thanks!
